pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -428,6 +518,11 @@ func (p *PoolManager) MarkVMAsReady(vm *kubevirt.VirtualMachine, latestPersisted
 	}
 	if len(macsToCommit) != 0 {
 		p.commitChangesToMacPoolMap(macsToCommit, vm, logger)
+
+		err = p.removeUpdatedMacsFromLegacyConfigMap(macsToCommit)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to update configMap for vm %s", vmFullName)
+		}
```

Comment:
Is  this needed ? we are already doing a migration at init with `initMacMapFromLegacyConfigMap`